src/main/java/org/jenkinsci/plugins/workflow/support/DefaultStepContext.java

```diff
@@ -61,7 +61,7 @@
         T value = doGet(key);
         if (key == EnvVars.class) {
             Run<?,?> run = get(Run.class);
-            EnvironmentAction a = run.getAction(EnvironmentAction.class);
+            EnvironmentAction a = run == null ? null : run.getAction(EnvironmentAction.class);
```

Comment:
We can get a null `Run` here now? How does that happen? I’m a bit nervous due to how many places assume you can never actually get a null `Run` from a `StepContext`.

Reply:
It shouldn't but it came up in tests once, so I bulletproofed it -- but that's likely some bad code that was now fixed.  Left it in because... better nullchecks, why not?

Reply:
Okiedokie. Just making sure it wasn't an expected result of a change.

Reply:
It _is_ correct to check for null here. Most places calling `get(Run.class)` do not need to, because `Run` was already declared part of the required context. Of course there are no production implementations of `FlowExecutionOwner` that are not based on `Run`, so this would never have come up. I suppose you hit this due to use of mocks.